src/plone/api/content.py

```diff
@@ -676,23 +676,30 @@ def get_parents(obj=None, predicate=None, interface=None):
 
     :param obj: [required] Object for which we want to get the parents.
     :type obj: Content object
-    :param predicate: Optional callable that takes an object and returns a boolean.
-        Used to filter the parents.
-    :type predicate: callable
+    :param interface_or_predicate: For backward compatibility - can be either an interface or predicate
+    :type interface_or_predicate: zope.interface.Interface or callable
     :param interface: Optional interface to filter the parents.
     :type interface: zope.interface.Interface
     :returns: List of parent objects, from immediate to site root.
     :rtype: list
     :Example: :ref:`content-get-parents-example`
-
     """
-    chain = aq_chain(aq_inner(obj))[1:]
+    # Get the parent chain but filter out non-contentish parents like Application and RequestContainer
+    chain = [obj for obj in aq_chain(aq_inner(obj))[1:] if isinstance(obj, DynamicType)]
 
     if interface is not None:
-        chain = (obj for obj in chain if interface.providedBy(obj))
+        chain = [parent for parent in chain if interface.providedBy(parent)]
```

Comment:
Please revert this change, it was good before because you were using an iterator and not a list comprehension.